Documentation/Walkthrough.md

```diff
@@ -0,0 +1,206 @@
+# Getting started
+
+The definition and implementation of data and service contracts is the same as WCF. The major difference is in the definition of the host which is now based on ASP.NET Core, and the ceremony of how a service is exposed. The following is based on .NET 6, but the same steps apply to other versions of .NET.
+
+## Defining the service
+
+### 1. Create an ASP.NET Core Empty application, this provides the host for the service.
+
+*Visual Studio:*
+ ![Project Template](images/empty-web-project.png)
+
+*Command Line:*
+
+```cli
+mkdir CoreWCFDemoServer
+dotnet new web -n CoreWCFDemoServer -o CoreWCFDemoServer
+```
+
+### 1. Add references to the CoreWCF Nuget Packages
+
+*Visual Studio:*
+
+Using the package Manager console, add: 
+
+- CoreWCF.Primitives
+- CoreWCF.Http
+
+![Package Manager Console](images/nuget.png)
+
+*Command Line:*
+
+Edit the project file and add:
+
+```xml
+<ItemGroup>
+  <PackageReference Include="CoreWCF.Primitives" Version="1.0.0" />
+  <PackageReference Include="CoreWCF.Http" Version="1.0.0" />
+</ItemGroup>
+```
+### 1. Create the Service Contract and Data Contract definitions
+
+These are defined the same as with WCF. When modernizing projects, this code can remain largely unchanged.
+
+**File: IEchoService.cs**
+
+```csharp
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.Serialization;
+using CoreWCF;
+
+namespace CoreWCfDemoServer
+{
+    [DataContract]
+    public class EchoFault
+    {
+        [AllowNull]
+        private string _text;
+
+        [DataMember]
+        [AllowNull]
+        public string Text
```

Comment:
Why not use an auto property?

Reply:
I think I stole this code from the samples.